Client/Frontend/Widgets/PhotonActionSheet.swift

```diff
@@ -657,20 +667,46 @@ private class PhotonActionSheetCell: UITableViewCell {
         accessibilityLabel = action.title
         selectionStyle = action.handler != nil ? .default : .none
 
-        if let iconName = action.iconString, let image = UIImage(named: iconName)?.withRenderingMode(.alwaysTemplate) {
-            statusIcon.sd_setImage(with: action.iconURL, placeholderImage: image, options: []) { (img, err, _, _) in
-                if let img = img {
-                    self.statusIcon.image = img.createScaled(PhotonActionSheetUX.IconSize)
-                    self.statusIcon.layer.cornerRadius = PhotonActionSheetUX.IconSize.width / 2
+        if let iconName = action.iconString {
+            switch action.iconType {
+            case .Image:
+                let image = UIImage(named: iconName)?.withRenderingMode(.alwaysTemplate)
+                statusIcon.image = image
+                self.statusIcon.layer.cornerRadius = PhotonActionSheetUX.IconSize.width / 2
+                statusIcon.tintColor = self.tintColor
+            case .URL:
+                let image = UIImage(named: iconName)?.withRenderingMode(.alwaysTemplate)
+                statusIcon.sd_setImage(with: action.iconURL, placeholderImage: image, options: []) { (img, err, _, _) in
+                    if let img = img {
+                        self.statusIcon.image = img.createScaled(PhotonActionSheetUX.IconSize)
+                        self.statusIcon.layer.cornerRadius = PhotonActionSheetUX.IconSize.width / 2
+                    }
                 }
-            }
-            // When the iconURL is not nil we are most likely showing a profile picture.
-            // In that case we do not need a tint color. And make sure the image is sized correctly.
-            // This is for the sync profile button in the menu
-            if action.iconURL == nil {
+            case .TabsButton:
+                if tabButton == nil {
+                    tabButton = TabsButton()
+                }
+                let renderer = UIGraphicsImageRenderer(size: (tabButton?.bounds.size)!)
+                let image = renderer.image { ctx in
+                    tabButton?.drawHierarchy(in: (tabButton?.bounds)!, afterScreenUpdates: true)
```

Comment:
We should be able to just add the `tabButton` to the view hierarchy without having to turn it into an image. Similar to how the syncButton works.

Reply:
I had tried that initially and left it commented. The icon still doesn't show up so maybe I am missing something I don't know about